stabilizer/src/main/java/com/hazelcast/stabilizer/worker/selector/OperationSelector.java

```diff
@@ -0,0 +1,35 @@
+package com.hazelcast.stabilizer.worker.selector;
+
+import java.util.Random;
+
+/**
+ * Facility to select different operations based on probabilities. Each operations is represented as an enum item.
+ * <p/>
+ * Calling {@link #select()} method will select an operation according to the configured probabilities.
+ * <p/>
+ * This class does not give any thread-safety guarantees. It is strongly recommended to construct a new instance for each thread,
+ * at least to prevent contention on the random generator. Just use a single builder and call the
+ * {@link OperationSelectorBuilder#build()} method in each thread constructor.
+ *
+ * @param <T> enum of operations
+ */
+public class OperationSelector<T extends Enum<T>> {
+    private final Random random;
+    private final Object[] operations;
+
+    OperationSelector(Object[] operations) {
+        this.random = new Random();
```

Comment:
Personal flavour; do whatever you like. If an object can be initialized in the class constructor, I do that. It keeps the normal constructor cleaner and shorter. Especially when you have some initialization logic in the normal constructor, this can be a good way to get rid of things that don't to be there.